src/main/kotlin/com/theapache64/dde/main.kt

```diff
@@ -1,19 +1,21 @@
 package com.theapache64.dde
 
-import androidx.compose.desktop.Window
-import com.arkivanov.decompose.extensions.compose.jetbrains.rememberRootComponent
-import com.arkivanov.decompose.extensions.compose.jetbrains.rootComponent
+import androidx.compose.ui.window.singleWindowApplication
+import com.arkivanov.decompose.DefaultComponentContext
+import com.arkivanov.essenty.lifecycle.LifecycleRegistry
 import com.theapache64.dde.navigation.NavHostComponent
 import com.theapache64.dde.theme.DecomposeDesktopExampleTheme
 
 /**
  * Where all magic starts ;)
  */
-fun main() = Window(
+fun main() = singleWindowApplication(
     title = "Decompose Desktop Example"
 ) {
+    val lifecycle = LifecycleRegistry()
+    val root = NavHostComponent(DefaultComponentContext(lifecycle))
```

Comment:
It's [advised](https://arkivanov.github.io/Decompose/component/overview/#root-componentcontext-in-jetpackjetbrains-compose) to either create the root context and component outside of compose, or `remember` it.